mpyg321/MPyg123Player.py

```diff
@@ -51,6 +52,15 @@ def unmute(self):
         """Unmutes the player"""
         self.player.sendline("UNMUTE")
 
+    def toggle_mute(self):
+        """Mute or UnMute if playing"""
+        if self.status == PlayerStatus.PLAYING:
+            if self.is_muted:
+                self.unmute()
+            else:
+                self.mute()
+            self.is_muted = not self.is_muted        
```

Comment:
Make sure you also set the value of `is_muted` inside the `mute` and `unmute` functions.
Another way to handle this would be to set the value inside the `process_output_ext` function (before calling the event callbacks). However, that design could lead to race conditions (where you toggle the mute twice, but too fast, so that it will actually not toggle the second time). I'm not sure how critical it is.